pypeit/sensfunc.py

```diff
@@ -1,164 +1,383 @@
+                        or isinstance(self[key], table.Table):
+                    continue
+                _d[key] = self[key]
+            # Add the algorithm if it has been defined
+            if self.algorithm is not None:
+                _d['algorithm'] = self.algorithm
+
+        return d
 
+    @classmethod
+    def from_hdu(cls, hdu, hdu_prefix=None, chk_version=True):
+        """
+        Instantiate the object from an HDU extension.
+
+        This overrides the base-class method, essentially just to handle the
+        fact that the 'TELLURIC' extension is not called 'MODEL'.
+
+        Args:
+            hdu (`astropy.io.fits.HDUList`_, `astropy.io.fits.ImageHDU`_, `astropy.io.fits.BinTableHDU`_):
+                The HDU(s) with the data to use for instantiation.
+            hdu_prefix (:obj:`str`, optional):
+                Maintained for consistency with the base class but is
+                not used by this method.
+            chk_version (:obj:`bool`, optional):
+                If True, raise an error if the datamodel version or
+                type check failed. If False, throw a warning only.
+        """
+        # Run the default parser to get most of the data. This correctly parses
+        # everything except for the Telluric.model data table.
+        d, version_passed, type_passed, parsed_hdus = super()._parse(hdu, allow_subclasses=True)
+        if not type_passed:
+            msgs.error('The HDU(s) cannot be parsed by a {0} object!'.format(cls.__name__))
+        if not version_passed:
+            _f = msgs.error if chk_version else msgs.warn
+            _f('Current version of {0} object in code (v{1})'.format(cls.__name__, cls.version)
+               + ' does not match version used to write your HDU(s)!')
+
+        # Load the telluric model, if it exists
+        if 'TELLURIC' in [h.name for h in hdu]:
+            # Instantiate the Telluric model from the header, if it exists
+            # TODO: I don't like this, but this is the fastest work around
+            d['telluric'] = telluric.Telluric.from_hdu(hdu['TELLURIC'], ext_pseudo='MODEL',
+                                                       chk_version=chk_version)
+        # Return the constructed object
+        return super().from_dict(d=d)
+
+    def compute_zeropoint(self):
+        """
+        Dummy method overloaded by subclasses
         """
         pass
-        return None, None
 
     def run(self):
+        """
+        Execute the sensitivity function calculations.
+        """
         # Compute the sensitivity function
-        self.meta_table, self.out_table = self.compute_zeropoint()
+        self.compute_zeropoint()
+
         # Extrapolate the zeropoint based on par['extrap_blu'], par['extrap_red']
-        self.wave_zp, self.zeropoint = self.extrapolate(samp_fact = self.par['samp_fact'])
+        self.wave, self.zeropoint = self.extrapolate(samp_fact=self.par['samp_fact'])
```

Comment:
Given all the different wavelengths to keep track of in this routine, I'd appreciate it if you revert to the original naming convention that I established to keep them straight, i.e. wave_zp, wave_zp_splice, etc. There are a lot of wavelength arrays to keep track of here given all the different modes in which this routine needs to work (i.e. multislit, echelle, multisit with splicing, etc.

Reply:
Right, and this name change was my attempt to clean that up.  The primary issue is that the datamodel (i.e., the things written to the output file) include `wave`, `zeropoint`, and `throughput`, so I wanted those to match.  So instead of having to introduce bespoke code that parsed `self.wave_zp` into the `WAVE` output extension, I changed the name of variable.  The only other name I had to change was `self.wave` -> `self.wave_cnts`.  So now the various array groupings are:

 - `self.wave_cnts, self.counts, self.counts_ivar, self.counts_mask`
 - `self.wave, self.zeropoint, self.throughput`
 - `self.wave_slice, self.zeropoint_slice, self.throughput_slice`

This feels cleaner.

Reply:
Okay, yes I see the point now with the data model. I'm fine with this naming convention, since it would be superfluous to name wave wave_zp in the data model where there is only a single wavelength.